server/src/auth/auth.service.ts

```diff
@@ -67,4 +74,66 @@ export class AuthService {
       updatedAt: user.updatedAt,
     };
   }
+
+  async loginService(token: string): Promise<any> {
+    try {
+      const ticket = await this.googleClient.verifyIdToken({
+        idToken: token,
+        audience: process.env.GOOGLE_CLIENT_ID,
+      });
+
+      const payload = ticket.getPayload();
+      const email = payload?.email;
+
+      if (!email) {
+        throw new ForbiddenException('Invalid Google token');
+      }
+
+      const existingUser = await this.prisma.user.findUnique({
+        where: {
+          email,
+        },
+      });
+
+      if (!existingUser) {
+        const hashpass = await argon.hash(uid(16));
+        const newUser = await this.prisma.user.create({
+          data: {
+            email,
+            password: hashpass,
+            randomToken: uid(16),
+          },
+          select: {
+            id: true,
+            email: true,
+            randomToken: true,
+            createdAt: true,
+            updatedAt: true,
+          },
+        });
+
+        return {
+          user: {
+            id: newUser.id,
+            email: newUser.email,
+            randomToken: newUser.randomToken,
+            createdAt: newUser.createdAt,
+            updatedAt: newUser.updatedAt,
+          },
+        };
+      }
+
+      return {
```

Comment:
before return generate a new random token for the user like lign 61

Reply:
filled the service with bad infos caus i will do it on another branch caus it involve some change on my google cloud etc so another functionnality so on another branch